packages/replay-next/src/utils/array.ts

```diff
@@ -98,6 +99,92 @@ export function findInsertIndex<T>(
   return lowIndex;
 }
 
+export function findSliceIndices<ItemType, TargetType>(
```

Comment:
:question: what's all this used for?

Reply:
I wrote it to more efficiently splice out the network events that were relevant to a specific test, in a recording that contained multiple tests:
https://github.com/replayio/devtools/pull/9292/files#diff-de0bf61f1db9a5c0ca715ab5158845281026d0afd81cdced087f540e38337732R288-R324